### DISCUSSION TITLE
A couple of tools fail on work laptop with ESET while Windhawk is running

### DISCUSSION THREAD

**Discussion by vitaminj** [ramensoftware/windhawk#14] [General]
I have a problem with a couple of tools I use while developing, which seem to fail on launch with Windhawk running, even if I disable all plugins. They work fine on my personal laptop, but on my work laptop (admin rights only on request, ESET Endpoint Security 9.1) they fail.

The first is running chrome webdriver tests. This can be replicated with this package.json and running `npm install` and then `npm start`:
```
{
  "scripts": {
    "start": "echo require('chromedriver'); new (require('selenium-webdriver').Builder)().forBrowser('chrome').build(); | node"
  },
  "dependencies": {
    "chromedriver": "104.0.0",
    "selenium-webdriver": "2.52.0"
  }
}
```
Error: Server terminated early with status 3221226505
Trying to look through the code, chromedriver.exe launches fine, but as soon as the node host tries to get session info from it, it dies.
What can I do to collect logs as to what breaks here? There's nothing in the ESET logs to show what it is blocking at the time, although I can see it blocking Windhawk hooking into its own processes earlier in the log. Windhawk generally works otherwise though.

The second failure is something within the sequence of processes that makes a squirrel installer for an electron app:
Failed with exit code: 3221225477
I can try to get logs again?

Both of the above work fine with Windhawk closed.

**Comment by m417z** [comment]
> This can be replicated with this package.json and running npm install and then npm start

Worked for me, see screenshot below.

> What can I do to collect logs as to what breaks here?

Can you capture and upload a crash dump?
You can use ProcDump. Run `procdump64.exe -mp -e node.exe` and trigger the crash. The dump will be written to a file, the details will be printed in the ProcDump console window. If another process crashes, not node.exe, replace `node.exe` with the process name. You can find out which process crashes in the Event Viewer. Look here for instructions: [How do I find event logs when a program crashes?](http://www.cyberlink.com/support/faq-content.do?id=10449)

> The second failure is something within [...] a squirrel installer

You can try the above for this case, too.

> Both of the above work fine with Windhawk closed.

I plan on adding an option to exclude processes from Windhawk. In this case, I suspect that the failures have something to do with ESET, and incompatibility with security software is something I have little control over, but if only specific programs are affected, perhaps process exclusion might be a workaround which is good enough.

![image](https://user-images.githubusercontent.com/4129781/187303320-99440120-6843-42a1-afbf-a4d3943e3e5b.png)

**Comment by m417z** [comment]
With Windhawk v1.0, it's now possible to exclude processes in Windhawk. Please try it and let me know whether it helps.
